internal/test/wsu/wsu_test.go

```diff
@@ -55,79 +53,235 @@ var (
+	require.NotEmptyf(t, playbookPath, "WSU_PATH environment variable not set")
+
+	// Get URL of last released wmcb
+	wmcbPinnedURL, err := getLatestReleasedArtifactURL("wmcb.exe")
+	require.NoError(t, err, "Could not get WMCB url")
+
+	// Check that we used the pinned version
+	require.Contains(t, ansibleOutput, wmcbPinnedURL, "Pinned version was not used in WSU")
+
+	// Download the wmcb.exe and check the SHA
+	resp, err := http.Get(wmcbPinnedURL)
+	require.NoErrorf(t, err, "Could not download wmcb.exe from %s", wmcbPinnedURL)
+	defer resp.Body.Close()
+	require.Equal(t, http.StatusOK, resp.StatusCode, "github api returned non OK status code: %v", resp)
+	body, err := ioutil.ReadAll(resp.Body)
+	require.NoError(t, err)
+	wmcbPinnedURLSha256 := sha256.Sum256(body)
+
+	// Ensure the WSU used the correct WMCB via SHA256
+	wmcbPath := wmcbExeRegex.FindString(ansibleOutput)
+	require.NotEmpty(t, wmcbPath)
+	wmcbContents, err := ioutil.ReadFile(wmcbPath)
+	downloadedWMCBSha := sha256.Sum256(wmcbContents)
+	assert.Equal(t, wmcbPinnedURLSha256, downloadedWMCBSha, "Downloaded WMCB has different SHA256 than expected")
+}
+
+// testBuiltWMCB tests that the WMCB was built, and not downloaded
+func testBuiltWMCB(t *testing.T, ansibleOutput string) {
+	require.Contains(t, ansibleOutput,
+		"CGO_ENABLED=0 GO111MODULE=on GOOS=windows go build -o wmcb.exe  "+
+			"github.com/openshift/windows-machine-config-operator/cmd/bootstrapper")
+}
+
 // TestWSU creates a Windows instance, runs the WSU, and then runs a series of tests to ensure all expected
 // behavior was achieved. The following environment variables must be set for this test to run: KUBECONFIG,
 // AWS_SHARED_CREDENTIALS_FILE, ARTIFACT_DIR, KUBE_SSH_KEY_PATH, WSU_PATH, CLUSTER_ADDR
 func TestWSU(t *testing.T) {
 	require.NotEmptyf(t, playbookPath, "WSU_PATH environment variable not set")
 	require.NotEmptyf(t, clusterAddress, "CLUSTER_ADDR environment variable not set")
+	require.GreaterOrEqual(t, vmCount, 1, "Expected one or more VMs")
+
+	wmcbPinnedURL, err := getLatestReleasedArtifactURL("wmcb.exe")
+	require.NoError(t, err, "Could not get WMCB url")
 
-	for _, vm := range framework.WinVMs {
-		// Run the test suite twice, to ensure that the WSU can be run multiple times against the same VM
-		runWSUTestSuite(t, vm)
-		t.Run("Run the WSU against the same VM again", func(t *testing.T) {
-			runWSUTestSuite(t, vm)
+	// Run VM specific tests
+	for i, vm := range framework.WinVMs {
+		t.Run("VM "+strconv.Itoa(i), func(t *testing.T) {
+			testVM(t, i, vm, wmcbPinnedURL)
 		})
 	}
+
+	// Run cluster wide tests
+	t.Run("Pending CSRs were approved", testNoPendingCSRs)
 }
 
-// runWSUTestSuite runs the WSU test suite against the VM
-func runWSUTestSuite(t *testing.T, vm e2ef.WindowsVM) {
+// testVM runs the WSU against the given VM and runs the e2e test suite against that VM as well
+func testVM(t *testing.T, vmNum int, vm e2ef.WindowsVM, wmcbURL string) {
```

Comment:
This can be renamed to `runWSU` and `runTest`. `runWSU` will have the logic for picking whether to use the pinned to built WMCB.